### ISSUE TITLE
Check if Jenkins API can rotate secrets

### ISSUE THREAD

**Issue by Halamix2** [kyma-project/test-infra#5833]
**Description**

- Check if Jenkins API can rotate Jenkins secrets
- Check if we can update secrets from Jenkins pipeline

**Reasons**

We want to automatically update secrets in Jenkins form GCP Secret Manager.
Jenkins is not accessible from the outside, so the service inside our infrastructure needs to do the rotation

Issue estimated under #5832 

 - Parent: #5832

**Comment by dekiel** [comment]
I did try setup https://plugins.jenkins.io/gcp-secrets-manager-credentials-provider/ without success. A plugin requires to define environment variable for jenkins process to allow plugin populate credentials store with secrets from GCP secret manager and keep them up to date. This was not possible due to limitations of our jaas instance. I have still ongoing discussion with jaas team to make sure there is no workaround for this problem.

Apart from the gcp secrets manager plugin I was able to update credentials in a Jenkins store by calling credential plugin API REST endpoint. API calls are made from Jenkins pipeline and allow reading and updating credentials data.
